src/cpp/src/sequence_group.hpp

```diff
@@ -592,7 +598,9 @@ class SequenceGroup {
         } else if (m_sampling_params.is_greedy_decoding() || m_sampling_params.is_multinomial()) {
             // We can stream only when one sequence is returned and we don't use stop strings that would be excluded from the output
             // (after stop string is detected its tokens are already sent)
-            if (num_total_seqs() == 1&& (m_sampling_params.stop_strings.empty() || m_sampling_params.include_stop_str_in_output)) {
+            if (num_total_seqs() == 1 &&
+                (m_sampling_params.stop_strings.empty() || m_sampling_params.include_stop_str_in_output) &&
+                !m_sampling_params.is_speculative_decoding()) {
```

Comment:
could you please clarify / add comments why do we need this check?

Reply:
In speculative decoding we will produce several tokens simultaneously, so we can not push_output token by token

Reply:
Fixed